src/main/java/com/tabnine/lifecycle/BinaryNotificationsLifecycle.kt

```diff
@@ -11,52 +11,56 @@ import com.tabnine.binary.requests.notifications.NotificationOptions
-                                binaryRequestFacade.executeRequest(
-                                    NotificationActionRequest(
-                                        binaryNotification.id,
-                                        o.key,
-                                        binaryNotification.message,
-                                        binaryNotification.notificationType,
-                                        o.actions
+                        binaryNotification.options?.stream()?.findFirst()?.ifPresent { o: NotificationOptions ->
+                            notification.addAction(object : AnAction(o.key) {
+                                override fun actionPerformed(e: AnActionEvent) {
+                                    binaryRequestFacade.executeRequest(
+                                        NotificationActionRequest(
+                                            binaryNotification.id,
+                                            o.key,
+                                            binaryNotification.message,
+                                            binaryNotification.notificationType,
+                                            o.actions
+                                        )
                                     )
-                                )
-                                if (o.actions?.contains(OPEN_HUB_ACTION)?: false){ 
-                                    actionVisitor.openHub() 
+                                    if (o.actions?.contains(OPEN_HUB_ACTION) ?: false) {
+                                        actionVisitor.openHub()
+                                    }
+                                    notification.expire()
                                 }
-                                notification.expire()
-                            }
-                        })
-                    }
+                            })
+                        }
 
-                    Notifications.Bus.notify(notification)
-                    binaryRequestFacade.executeRequest(NotificationShownRequest(binaryNotification.message))
-                    PropertiesComponent.getInstance().setValue(storageKey(binaryNotification.id), true)
+                        Notifications.Bus.notify(notification)
+                        binaryRequestFacade.executeRequest(NotificationShownRequest(binaryNotification.message))
+                        PropertiesComponent.getInstance().setValue(storageKey(binaryNotification.id), true)
+                    }
                 }
-            }
-        }, BINARY_NOTIFICATION_POLLING_INTERVAL, BINARY_NOTIFICATION_POLLING_INTERVAL)
+            },
+            BINARY_NOTIFICATION_POLLING_INTERVAL, BINARY_NOTIFICATION_POLLING_INTERVAL
+        )
     }
 
-    private fun storageKey(id: String?) = "tabnine-notifications-shown-${id}"
-}
\ No newline at end of file
+    private fun storageKey(id: String?) = "tabnine-notifications-shown-$id"
```

Comment:
Hard to tell if this is the only thing that changed here. I think it's better to separate format changes into a dedicated PR.
Also, this looks like a bug for something that was never tested? Did you test it now?

Reply:
Nothing changed in that file, I only ran `ktlint` to format the whole files

Reply:
Looks like the curly braces around the id were deleted.